Paintroid/src/main/java/org/catrobat/paintroid/dialog/PermissionInfoDialog.java

```diff
@@ -37,18 +40,21 @@
 	private static final String PERMISSION_TYPE_KEY = "permissionTypeKey";
 	private static final String PERMISSIONS_KEY = "permissionsKey";
 	private static final String REQUEST_CODE_KEY = "requestCodeKey";
+	private static final String PERMANENT_DENIAL_KEY = "permanentDenialKey";
 
 	private int requestCode;
 	private String[] permissions;
 	private PermissionType permissionType;
+	private boolean permanent;
 
-	public static PermissionInfoDialog newInstance(PermissionType permissionType, String[] permissions, int requestCode) {
+	public static PermissionInfoDialog newInstance(PermissionType permissionType, String[] permissions, int requestCode, boolean permanent) {
```

Comment:
Don't add boolean parameters. This increases cyclomatic complexity, and leads to switches in UI classes and code duplication.
Keep UI classes as dumb as possible and move any logic outside.

Reply:
See https://www.martinfowler.com/bliki/FlagArgument.html for an article by Martin Fowler on the topic of boolean arguments / flag arguments